docs/runtimes/create-and-use-functions.ipynb

```diff
@@ -0,0 +1,550 @@
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8ab1e780",
+   "metadata": {},
+   "source": [
+    "<a id=\"using-set_function\"></a>\n",
+    "### Using set_function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c878374d",
+   "metadata": {},
+   "source": [
+    "The MLRun project object has a method called {py:meth}`~mlrun.projects.MlrunProject.set_function` which is a one-size-fits-all way of creating an MLRun function. This method accepts a variety of sources including Python files, Jupyter notebooks, Git repos, and more.\n",
+    "\n",
+    "``` {admonition} Note\n",
+    "The return value of `set_function` will be your MLRun function. You can immediately run it or apply additional configuration like resources, scaling, etc. See [**Customizing Functions**](#customizing-functions) for more details.\n",
+    "```\n",
+    "\n",
+    "When using `set_function`, there are a number of common parameters across all function types and creation scenarios. Consider the following example:\n",
+    "\n",
+    "```python\n",
+    "fn = project.set_function(\n",
+    "    name=\"my-function\", tag=\"latest\", func=\"my_function.py\",\n",
+    "    image=\"mlrun/mlrun\", kind=\"job\", handler=\"train_model\",\n",
+    "    requirements=[\"pandas==1.3.5\"], with_repo=True\n",
+    ")\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bb9cebf9",
+   "metadata": {},
+   "source": [
+    "- **Name:** Name of your MLRun function within the given project. This will be displayed in the MLRun UI as well as the Kubernetes pod.\n",
+    "\n",
+    "- **Tag:** Tag for your function (much like a Docker image). Ommiting this parameter will default to `latest`.\n",
+    "\n",
+    "- **Func:** What to run with the MLRun function. This can be a number of things including files (`.py`, `.ipynb`, `.yaml`, etc.), URIs (`hub://` prefixed function marketplace URI, `db://` prefixed MLRun DB URI), existing MLRun Function objects, or `None` (for current `.ipynb` file).\n",
+    "\n",
+    "- **Image:** Docker image to use when containerizing the piece of code. If you also specify the `requirements` parameter to build a new Docker image, the `image` parameter will be used as the base image.\n",
+    "\n",
+    "- **Kind:** Runtime the MLRun function uses. See [**Kinds of functions (runtimes)**](../concepts/functions-overview.html) for which batch and real-time runtimes are supported.\n",
+    "\n",
+    "- **Handler:** Default function handler to invoke (e.g. a Python function within your script). This handler can also be overriden when executing the function.\n",
+    "\n",
+    "- **Requirements:** Additional Python dependencies needed for the function to run. Using this parameter will result in a new Docker image (using the `image` parameter as a base image). This can be a list of Python dependencies or a path to a `requirements.txt` file.\n",
```

Comment:
"- **requirements:** Additional Python dependencies needed for the function to run. Using this parameter results in a new Docker image (using the `image` parameter as a base image). This can be a list of Python dependencies or a path to a `requirements.txt` file.\n",